### PR TITLE
feat: Service client release notes

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

1734

## Description of changes

The original [PR](https://github.com/awslabs/aws-sdk-swift/pull/1780) was reverted due to unexpected build failures.

This PR contains every change from reverted PR + changed behavior for decoding `releaseNotes`, treating it as nullable now. It adds a test case for this as well.

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Models/Features.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        return try JSONDecoder().decode([String: String].self, from: fileContents)
+    }
+}
+
+struct Features: Decodable {
+    let features: [Feature]
+}
+
+struct Feature: Decodable {
+    let releaseNotes: String?
```
Comment (sichanyoo) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Models/Features.swift:39]: Make this optional !!

File: AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        let releaseNotes = try builder.build()
+        let expected = """
+        ## What's Changed
+
+        **Full Changelog**: https://github.com/awslabs/aws-sdk-swift/compare/1.0.0...1.0.1
+        """
+        XCTAssertEqual(releaseNotes, expected)
+    }
+
+    func testNullReleaseNotesFieldGetsHandledWithoutError() throws {
```
Comment (sichanyoo) [inline AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift:179]: Test added for null `releaseNotes` case

**Review by sichanyoo — COMMENTED** [review]
Put comments on things added from reverted PR

File: AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                "featureType": "DOC_UPDATE",
+            }
+        }
+    }
+    """
+
+    // Feature with null releaseNotes field
+    private let feature4 = """
+    {
+        "releaseNotes": null,
```
Comment (jbelkins) [inline AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift:60]: This is how preview builds encode release notes: the `releaseNotes` key is present with a `null` value?

Do we need to test where the `releaseNotes` key is omitted entirely?

Comment (jbelkins) [reply AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift:60]: (I _think_ an autogenerated Swift decoder will decode either case to a `String?` value of `nil`.)

**Review by jbelkins — COMMENTED** [review]
Just a question on form of JSON

File: AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift (hunk 3 above)
Comment (sichanyoo) [reply AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift:60]: Yeah, it's `releaseNotes` key present but with a `null` value.

Comment (sichanyoo) [reply AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift:60]: And yeah I think in either case Swift decoder just assigns `nil` as well

**Review by jbelkins — APPROVED** [review]

**Pull request merged**